src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -372,7 +372,104 @@ protected void InitializeReferences()
             RegexRunner runner = Interlocked.Exchange(ref _runner, null) ?? CreateRunner();
             try
             {
-                return runner.Scan(this, input, beginning, beginning + length, startat, prevlen, quick, internalMatchTimeout);
+                runner.InitializeTimeout(internalMatchTimeout);
+                runner.runtext = input;
+                ReadOnlySpan<char> span = input.AsSpan(beginning, length);
```

Comment:
Doesn't need to be now, but we might consider deleting the above `if ((uint)length > (uint)input.Length)` check, as this `input.AsSpan(beginning, length)` is also going to validate the length (albeith the error message probably won't be as specific).

Reply:
Why null it out in InternalPerformScan at all? Why not just not it out in the callers that are setting it to non-null in the first place? e.g. pseudo-code
```C#
runner.runtext = input;
Match? match = runner.InternalPerformScan(...);
runner.runtext = null;
return match;
```
?